File: DSA/Ayush_Mishra_590014844/Day_17_Question_3.c
#include <stdio.h>

int integerSqrt(int n) {
    if (n < 2) return n;
    int left = 1, right = n, ans = 0;
    while (left <= right) {
        int mid = left + (right - left) / 2;
        if (mid <= n / mid) {
            ans = mid;
            left = mid + 1;
        } else right = mid - 1;
    }
    return ans;
}

int main() {
    int n;
    printf("Enter a positive number: ");
    scanf("%d", &n);
    printf("Integer square root: %d\n", integerSqrt(n));
    return 0;
}

File: DSA/Ayush_Mishra_590014844/Day_17_Question_3.py
def integer_sqrt(n):
    if n < 2:
        return n
    left, right = 1, n
    ans = 0
    while left <= right:
        mid = (left + right) // 2
        if mid * mid <= n:
            ans = mid
            left = mid + 1
        else:
            right = mid - 1
    return ans

n = int(input("Enter a positive number: "))
print("Integer square root:", integer_sqrt(n))
